modules/gdscript/gdscript_editor.cpp

```diff
@@ -1918,8 +1918,16 @@ static void _find_identifiers_in_base(const GDScriptCompletionContext &p_context
 				Ref<GDScript> script = base_type.script_type;
 				if (script.is_valid()) {
 					if (!_static && !p_only_functions) {
-						for (const Set<StringName>::Element *E = script->get_members().front(); E; E = E->next()) {
-							r_result.insert(E->get().operator String());
+						if (p_context.base && p_context.base->get_script_instance()) {
```

Comment:
`p_context.base` is the object in which you currently are, not the object on which you are completing.

Reply:
Uhm, weird, I tested it in an autoload and a node with a script and it does work in those cases.

Trying it in a loaded script or in the same script with `self` (or without) didn't work though. I'm not sure what should I do then.

Reply:
Thinking about it, the loaded script or using self wouldn't work because they don't have an object to instance the script in the first place, do they? I don't know, can we make a fake object to instance the script and try to get the properties from it?

Reply:
I think you can try to "cheat" for autoloads by just getting the autoload if it is a tool script. Otherwise, instancing an object with a tool script seems a bit dangerous.

Reply:
Yes, for this to work the script needs to be a tool script because it need to be executed in the editor. 

I've been thinking and I don't think the `self` case should work anyway, it needs to compile and execute code in a state where it can't compile its code because it's being edited.

I'm not sure about the preloaded script case though. In theory it should be possible although I have no idea how.

I think we should limit it to autoloads and nodes with a script (aka what's currently working 😅 )

Reply:
My comment is invalid here, sorry..